ext/QuadGKEnzymeExt.jl

```diff
@@ -84,10 +84,37 @@ function Base.:*(a::ClosureVector, b::Number)
     return b*a
 end
 
-function Enzyme.EnzymeRules.reverse(config, ofunc::Const{typeof(quadgk)}, dres::Active, cache, f::Union{Const, Active}, segs::Annotation{T}...; kws...) where {T}
+struct MixedClosureVector{F}
+    f::Base.RefValue{F}
+end
+
+function Base.:+(a::CV, b::CV) where {CV <: MixedClosureVector}
+    res = deepcopy(a)::CV
+    Enzyme.Compiler.recursive_accumulate(res, b, identity)
+    res
+end
+
+function Base.:-(a::CV, b::CV) where {CV <: MixedClosureVector}
+    res = deepcopy(a)::CV
+    Enzyme.Compiler.recursive_accumulate(res, b, x->-x)
+    res
+end
+
+function Base.:*(a::Number, b::CV) where {CV <: MixedClosureVector}
+    # b + (a-1) * b = a * b
+    res = deepcopy(b)::CV
+    Enzyme.Compiler.recursive_accumulate(res, b, x->(a-1)*x)
```

Comment:
```suggestion
    res = Enzyme.make_zero(b)::CV
    Enzyme.Compiler.recursive_accumulate(res, b, x->a*x)
```
This addresses @stevengj's concern in https://github.com/JuliaMath/QuadGK.jl/pull/110#issuecomment-2323012377 and can perhaps be ported to the ClosureVector equivalent too